packages/admin/blocks-admin/src/blocks/helpers/createCompositeBlockTextField.tsx

```diff
@@ -0,0 +1,27 @@
+import { Field, FinalFormInput } from "@comet/admin";
+import * as React from "react";
+import { FormProps } from "react-final-form";
+
+import { BlocksFinalForm } from "../../form/BlocksFinalForm";
+import { createCompositeSetting } from "./composeBlocks/createCompositeSetting";
+
+interface Options {
+    defaultValue: string;
+    displayName?: string;
+    fieldProps?: FormProps<Record<"value", string>, Record<"value", string>>;
+}
+
+export function createCompositeBlockTextField({ defaultValue, displayName, fieldProps }: Options) {
+    return createCompositeSetting<string>({
+        defaultValue,
+        AdminComponent: ({ state, updateState }) => (
+            <BlocksFinalForm<{ value: typeof state }>
+                {...fieldProps}
+                onSubmit={({ value }) => updateState(value ?? "")}
+                initialValues={{ value: state || undefined }}
+            >
+                <Field fullWidth name="value" type="text" label={displayName} component={FinalFormInput} />
```

Comment:
@jamesricky what do you think about adding support for multiline here? This would require changing the `component` right?

Reply:
@jomunker you could use TextField here, like in #1678